src/components/metagoofil/metagoofil.tsx

```diff
@@ -1,163 +1,244 @@
-    "file types: pdf, doc, xls, ppt, docx, pptx, xlsx. \n\nOptions for the tool can be found at: https://www.kali.org/tools/metagoofil/\n\n" +
-    "Using the tool:\n" +
-    "Step 1: Enter a website URL for the tool to search.\n" +
-    "       Eg: kali.org\n\n" +
-    "Step 2: Enter the desired number of results.\n" +
-    "       Eg: 100\n\n" +
-    "Step 3: Enter the limit for the number of files to be downloaded\n" +
-    "       Eg: 25\n\n" +
-    "Step 4: Enter the file type name to be extracted.\n" +
-    "       Eg: pdf\n\n" +
-    "Step 5: Click scan to commence the Metagoofil operation.\n\n" +
-    "Step 6: View the Output block below to view the results of the tool's execution.";
-
-interface FormValues {
-    webname: string;
-    searchmax: string;
-    filelimit: string;
-    filetype: string;
-    filepath: string;
+/**
+ * Represents the form values for the Metagoofil component.
+ */
+interface FormValuesType {
+    webName: string;
+    searchMax: string;
+    fileLimit: string;
+    fileType: string;
+    filePath: string;
 }
 
-export function Metagoofil() {
-    const [loading, setLoading] = useState(false);
-    const [output, setOutput] = useState("");
-    const [pid, setPid] = useState("");
-    const [customconfig, setCustomconfig] = useState(false);
-    const [downloadconfig, setDownloadConfig] = useState(false);
+/**
+ * The Metagoofil component.
+ * @returns The Metagoofil component.
+ */
+function Metagoofil() {
+    //Component State Variables.
+    const [loading, setLoading] = useState(false); // State variable to indicate loading state.
+    const [output, setOutput] = useState(""); // State variable to store the output of the command execution.
+    const [pid, setPid] = useState(""); // State variable to store the process ID of the command execution.
+    const [customConfig, setCustomConfig] = useState(false); //State variable to indicate if custom configuration is enabled
+    const [downloadConfig, setDownloadConfig] = useState(false); //State variable to indicate if download configuration is enabled
+    const [isCommandAvailable, setIsCommandAvailable] = useState(false); // State variable to check if the command is available.
+    const [loadingModal, setLoadingModal] = useState(true); // State variable that indicates if the modal is opened.
+    const [opened, setOpened] = useState(!isCommandAvailable); // State variable to indicate loading state of the modal.
+    const [allowSave, setAllowSave] = useState(false); // State variable to indicate if saving is allowed
+    const [hasSaved, setHasSaved] = useState(false); // State variable to indicate if the output has been saved
 
+    // Component Constants.
+    const title = "Metagoofil"; // Title of the component.
+    const description =
+        "Metagoofil is an information gathering tool designed for extracting metadata of public documents (pdf,doc,xls,ppt,docx,pptx,xlsx) belonging to a target company."; // Description of the component.
```

Comment:
I know this is from Kali but I reckon this reads better:
```suggestion
        "Metagoofil is an information gathering tool designed for extracting the metadata of public documents (pdf,doc,xls,ppt,docx,pptx,xlsx) that belong to a target company."; // Description of the component.
```